app/code/Magento/Store/Model/StoreCreateManagement.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Magento\Store\Model;
+
+use Magento\Store\Api\StoreCreateManagementInterface;
+
+class StoreCreateManagement implements StoreCreateManagementInterface
+{
+    /**
+     * @var \Magento\Store\Model\WebsiteFactory
+     */
+    private $websiteFactory;
+
+    /**
+     * @var \Magento\Store\Model\GroupFactory
+     */
+    private $groupFactory;
+
+    /**
+     * @var \Magento\Store\Model\StoreFactory
+     */
+    private $storeFactory;
+
+    /**
+     *
+     * @param \Magento\Store\Model\WebsiteFactory     $websiteFactory
+     * @param \Magento\Store\Model\GroupFactory       $groupFactory
+     * @param \Magento\Store\Model\StoreFactory       $storeFactory
+     */
+    public function __construct(
+        \Magento\Store\Model\WebsiteFactory $websiteFactory,
+        \Magento\Store\Model\GroupFactory $groupFactory,
+        \Magento\Store\Model\StoreFactory $storeFactory
+    ) {
+        $this->websiteFactory = $websiteFactory;
+        $this->groupFactory = $groupFactory;
+        $this->storeFactory = $storeFactory;
+    }
+
+    /**
+     *
+     * @param  array $data
+     * @return \Magento\Store\Model\Store
+     */
+    public function create($data)
+    {
+        /** @var \Magento\Store\Model\Store $storeModel */
+        $storeModel = $this->storeFactory->create();
+        $storeModel->setData($data);
+
+        /** @var \Magento\Store\Model\Group $groupModel */
+        $groupModel = $this->groupFactory->create()
+            ->load($storeModel->getGroupId());
+
+        $storeModel->setWebsiteId($groupModel->getWebsiteId());
+        $storeModel->save();
+
+        return $storeModel;
+    }
+
+    /**
+     *
+     * @return int
+     */
+    public function getDefaultGroupId()
```

Comment:
Please extract this method to a separate model: `Magento\Store\Model\GetDefaultStoreGroup::execute`

We are trying to keep classes as granular as possible to simplify the extensibility and possible further move/replace of the class.

Also, please add strict types to all the method signatures

```suggestion
    public function getDefaultGroupId(): int
```